WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -88,6 +88,10 @@ def index
 
     if @display == 'admin'
       @competitions = @competitions.includes(:delegates, :delegate_report)
+
+      if params[:delegate].present?
+        @competitions = @competitions.select { |c| c.delegates.map(&:id).include?(params[:delegate].to_i) }
```

Comment:
@jfly had a better idea, I think: to first select the delegate and then his/her competitions, avoiding the need to loop through every competition. See new commit